pkg/controller/operators/operatoconditiongenerator_controller.go

```diff
@@ -0,0 +1,126 @@
+package operators
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	operatorsv1 "github.com/operator-framework/api/pkg/operators/v1"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/operators/decorators"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/ownerutil"
+)
+
+// OperatorReconciler reconciles a Operator object.
+type OperatorConditionGenReconciler struct {
```

Comment:
Maybe `OperatorConditionGenerationController`? Calling it a "condition reconciler" is a bit misleading, since it's not technically watching conditions.